src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/runInTerminalTool.ts

```diff
@@ -294,13 +301,47 @@ export class RunInTerminalTool extends Disposable implements IToolImpl {
 				}
 				const execution = new BackgroundTerminalExecution(toolTerminal.instance, xterm, command);
 				RunInTerminalTool._backgroundExecutions.set(termId, execution);
-				const resultText = (
+				// Poll for output until the terminal is idle or 20 seconds have passed
+				outputAndIdle = await this._pollForOutputAndIdle(execution, false, token);
+				if (!outputAndIdle.idle) {
+					const chatModel = invocation.context?.sessionId && this._chatService.getSession(invocation.context?.sessionId);
+					if (chatModel instanceof ChatModel) {
+						const request = chatModel.getRequests().at(-1);
+						if (request) {
+							let requestMorePolling = false;
+							const waitPromise = new Promise<void>(resolve => {
+								const part = new ChatElicitationRequestPart(
+									new MarkdownString(localize('poll.terminal.waiting', "Continue waiting for `{0}` to finish?", command)),
+									new MarkdownString(localize('poll.terminal.polling', "Copilot will continue to poll for output to determine when the terminal becomes idle for up to 2 minutes.")),
+									'',
+									localize('poll.terminal.accept', 'Yes'),
+									localize('poll.terminal.reject', 'No'),
+									async () => {
+										requestMorePolling = true;
+										resolve();
+									},
+									async () => {
+										requestMorePolling = false;
+										resolve();
+									}
+								);
+								chatModel.acceptResponseProgress(request, part);
+							});
+							await waitPromise;
+							if (requestMorePolling) {
+								outputAndIdle = await this._pollForOutputAndIdle(execution, true, token);
+							}
+						}
+					}
+				}
+				let resultText = (
 					didUserEditCommand
 						? `Note: The user manually edited the command to \`${command}\`, and that command is now running in terminal with ID=${termId}`
 						: didToolEditCommand
 							? `Note: The tool simplified the command to \`${command}\`, and that command is now running in terminal with ID=${termId}`
 							: `Command is running in terminal with ID=${termId}`
 				);
+				resultText += outputAndIdle.idle ? `\n\nTerminal became idle with output:\n${outputAndIdle.output}` : `\n\nTerminal is still running, with output:\n${outputAndIdle.output}`;
```

Comment:
I think "the command" over "terminal" would be clearer. Also it might not be clear what "idle" means?